mne/rank.py

```diff
@@ -211,24 +226,165 @@ def _get_rank_sss(inst):
-            elif len(max_info) == 0:
-                raise ValueError(
-                    'Did not find any SSS record. You should use data-based '
-                    'rank estimate instead')
-    if len(max_infos) > 0:
-        max_info = max_infos[0]
+    proc_info = info.get('proc_history', [])
+    if len(proc_info) > 1:
+        logger.info('Found multiple SSS records. Using the first.')
+    if len(proc_info) == 0 or 'max_info' not in proc_info[0] or \
+            'in_order' not in proc_info[0]['max_info']['sss_info']:
+        raise ValueError(
+            'Did not find any SSS record. You should use data-based '
+            'rank estimate instead')
+    proc_info = proc_info[0]
+    max_info = proc_info['max_info']
+    inside = max_info['sss_info']['in_order']
+    nfree = (inside + 1) ** 2 - 1
+    nfree -= (len(max_info['sss_info']['components'][:nfree]) -
+              max_info['sss_info']['components'][:nfree].sum())
+    return nfree
+
+
+def _info_rank(info, ch_type, picks, rank):
+    if ch_type == 'meg' and rank != 'full':
+        try:
+            return _get_rank_sss(info)
+        except ValueError:
+            pass
+    return len(picks)
+
+
+def _compute_rank_int(inst, scalings=None, rank=None, info=None):
+    """Wrap compute_rank but yield an int."""
+    # XXX eventually we should unify how channel types are handled
+    # so that we don't need to do this, or we do it everywhere.
+    # Using pca=True in compute_whitener might help.
+    return sum(compute_rank(inst, scalings, rank, info).values())
+
+
+def compute_rank(inst, scalings=None, rank=None, info=None):
+    """Compute the rank of data or noise covariance.
+
+    This function will normalize the rows of the data (typically
+    channels or vertices) such that non-zero singular values
+    should be close to one.
+
+    Parameters
+    ----------
+    inst : instance of Raw, Epochs, or Covariance
+        Raw measurements to compute the rank from or the covariance.
+    scalings : dict | None (default None)
+        Defaults to ``dict(mag=1e15, grad=1e13, eeg=1e6)``.
+        These defaults will scale different channel types
+        to comparable values.
+    rank : int | None | dict | 'info' | 'full'
+        This controls the rank computation that can be read from the
+        measurement info or estimated from the data. See ``Notes`` for
+        details. The default is ``'info'``.
+    info : instance of Info | None
+        The measurement info used to compute the covariance. It is
+        only necessary if inst is a Covariance object.
+
+    Returns
+    -------
+    rank : dict
+        Estimated rank of the data for each channel type.
+        To get the total rank, you can use ``sum(rank.values())``.
+
+    Notes
+    -----
+    The ``rank`` parameter can be:
+
+    :class:`python:int`
+        This value is used as the MEG rank. For other channel types,
+        rank is taken from ``info``.
```

Comment:
what if you pass just eeg data? I would suggest to say that this possibility is only possible if you have eeg or meg only. Now if you want to get the rank as int as returned by _compute_rank_int maybe we can suggest to pass rank as a dict(ch_type=int) in case of different channel types. WDYT?

Reply:
We have a backward compat problem because in `master` rank-as-int at least in `cov.py` means `dict(meg=rank)`